src/core/project.js

```diff
@@ -48,7 +67,7 @@ export function projectToJSON(p: Project): Compatible<Project> {
   return toCompat(COMPAT_INFO, p);
 }
 
-export function projectFromJSON(j: Compatible<any>): Project {
+export function projectFromJSON(j: Compatible<SupportedProject>): Project {
```

Comment:
I'm fine leaving this as `Compatible<any>` since as you noted, it could actually be anything. However, if it's not a project we'll get a runtime error. that's fine. and this should allow us to not need the cast in test code...